### PR TITLE
ML.NET doc updates

### LINKED ISSUE #2357: Cuda is not available for GPU Text classification
**System Information (please complete the following information):**
 - Model Builder Version (available in Manage Extensions dialog): 16.14.0

**Describe the bug**
For the Text Classification scenario... on the environment tab it shows that the machine has a Cuda compatible GPU. However, when training there is an error: "cuda is not available, please confirm you have a cuda-support gpu". 

**To Reproduce**
1. Select Text- classification scenario
2. Select GPU
3. Select data
4. Train

**Screenshots**
image

**Additional context**
image

### LINKED ISSUE #2359: Text-classification out of memory for GPU training
**System Information (please complete the following information):**
 - Model Builder Version (available in Manage Extensions dialog): 
 - Visual Studio Version

**Describe the bug**
- On which step of the process did you run into an issue:
- Clear description of the problem:

**To Reproduce**

**Expected behavior**

**Screenshots**
image

**Additional context**

### LINKED ISSUE #31779: ML.NET November 2022 Doc Updates
- Deep Learning (dotnet/machinelearning#6393)
  - [ ] Concepts - What is Deep Learning? (P0) 
  - [x] Tutorial - Text Classification (P0)
  - [ ] Tutorial - Sentence Similarity (P1)
- AutoML (#32011)
  - [x] Overview - Automated ML (P0)
    - What is AutoML?
      - What is FLAML?
    - What can I do with AutoML?
  - [ ] How-To - Use [Automated ML API](https://learn.microsoft.com/dotnet/machine-learning/how-to-guides/how-to-use-the-automl-api) (P0)
    - ColumnInference
    - Featurizer
    - Experiment
    - Sweepable API
    - Search Space API
- Model Builder
  - [x] Conceptual documentation up to date
    - [ ] [What is Model Builder and how does it work?](https://learn.microsoft.com/dotnet/machine-learning/automate-training-with-model-builder)
  - [ ] Tutorials up to date
    - [ ]  [Regression: Predict prices](https://learn.microsoft.com/dotnet/machine-learning/tutorials/predict-prices-with-model-builder)
    - [x] [Binary: Analyze sentiment](https://learn.microsoft.com/dotnet/machine-learning/tutorials/sentiment-analysis-model-builder)
    - [ ] [Multiclass: Classify health violations](https://learn.microsoft.com/dotnet/machine-learning/tutorials/health-violation-classification-model-builder)
    - [ ] [Image classification: Landsat image classification](https://learn.microsoft.com/dotnet/machine-learning/tutorials/image-classification-model-builder)
    - [ ] [OD: Detect traffic signs](https://learn.microsoft.com/dotnet/machine-learning/tutorials/object-detection-model-builder)
    - [ ] [Recommendation: movie recommendations](https://learn.microsoft.com/dotnet/machine-learning/tutorials/movie-recommendation-model-builder)
  - [x] All references point to VS2022
  - [x] Install docs up to date
- CLI
  - [x] All references point to latest version
  - [x] Install docs up to date

### PR DESCRIPTION
**Pull request by luisquintanilla**
Fixes #31779
Fixes dotnet/machinelearning-modelbuilder#2357
Fixes dotnet/machinelearning-modelbuilder#2359

This doc addresses changes for: 

- ML.NET AutoML API
- Text Classification tutorial
- Installation guides
- ML Scenarios
- Incorporate changes from #32065 

Preview Links:

- Installation guides:
  - [MB](https://review.learn.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/install-model-builder?branch=pr-en-us-32011&tabs=visual-studio-2022)
  - [MB GPU](https://review.learn.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/install-gpu-model-builder?branch=pr-en-us-32011)
  - [CLI](https://review.learn.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/install-ml-net-cli?branch=pr-en-us-32011&tabs=windows)
- AutoML
  - [ML.NET How-To](https://review.learn.microsoft.com/dotnet/machine-learning/how-to-guides/how-to-use-the-automl-api?branch=pr-en-us-32011)
  - [What is AutoML?](https://review.learn.microsoft.com/dotnet/machine-learning/automated-machine-learning-mlnet?branch=pr-en-us-32011)
- Scenarios
  - [ML.NET](https://review.learn.microsoft.com/en-us/dotnet/machine-learning/how-does-mldotnet-work?branch=pr-en-us-32011)
  - [Model Builder](https://review.learn.microsoft.com/en-us/dotnet/machine-learning/automate-training-with-model-builder?branch=pr-en-us-32011)
- [Text classification MB Tutorial](https://review.learn.microsoft.com/dotnet/machine-learning/tutorials/sentiment-analysis-model-builder?branch=pr-en-us-32011)
- [Deep Learning Conceptual article]()

### CODE REVIEW

**Comment by JakeRadMSFT** [comment]
https://review.learn.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/how-to-use-the-automl-api?branch=pr-en-us-32011#load-data

image


Why is GroupColumns false?

**Comment by JakeRadMSFT** [comment]
image
Should that be 80% in the comment?

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md [hunk 1]
```diff
@@ -60,63 +54,68 @@ Each row in the *wikipedia-detox-250-line-data.tsv* dataset represents a differe
...
+## Create a Model Builder config file
+
+When first adding a machine learning model to the solution it will prompt you to create an `mbconfig` file. The `mbconfig` file keeps track of everything you do in Model Builder to allow you to reopen the session.
+
+1. In **Solution Explorer**, right-click the *SentimentRazor* project, and select **Add** > **Machine Learning Model...**.
+1. In the dialog, name the Model Builder project **SentimentAnalysis**, and click **Add**.
+
 ## Choose a scenario
 
 ![Model Builder wizard in Visual Studio](../media/model-builder-scenarios.png)
```
Comment (zewditu) [inline docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:66]: Use updated scenario  page screenshot

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md [hunk 2]
```diff
@@ -60,63 +54,68 @@ Each row in the *wikipedia-detox-250-line-data.tsv* dataset represents a differe
...
 
 To train your model, you need to select from the list of available machine learning scenarios provided by Model Builder.
 
-1. In **Solution Explorer**, right-click the *SentimentRazor* project, and select **Add** > **Machine Learning Model...**.
-1. For this sample, the scenario is sentiment analysis. In the *scenario* step of the Model Builder tool, select the **Sentiment Analysis** scenario.
+1. For this sample, the task is text classification. In the *Scenario* step of the Model Builder extension, select the **Text classification** scenario.
```
Comment (zewditu) [inline docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:70]: I think if you do not have another item to say , you do not need `1`, 
when I saw I was expecting 2.... but it should not be 
image

**Comment by JakeRadMSFT** [comment]
image
RandomSearch looks like it's missing some formatting.

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md [hunk 3]
```diff
@@ -60,63 +54,68 @@ Each row in the *wikipedia-detox-250-line-data.tsv* dataset represents a differe
...
+
+## Select an environment
+
+Model Builder can run the training on different environments depending on the scenario that was selected.
+
+Select **Local (GPU)** as your environment and click the **Next step** button.
+
+> [!NOTE]
+> This scenario uses deep learning techniques which work best in GPU environments. If you don't have a GPU, choose the Local (CPU) environment but note that the expected time to train will be significantly longer. For more information on using GPUs with Model Builder, see the [GPU support in Model Builder guide](../how-to-guides/install-gpu-model-builder.md).
```
Comment (zewditu) [inline docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:79]: are you able to add the link for environment selection for further details here ?

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md [hunk 4]
```diff
@@ -60,63 +54,68 @@ Each row in the *wikipedia-detox-250-line-data.tsv* dataset represents a differe
...
+1. In **Solution Explorer**, right-click the *SentimentRazor* project, and select **Add** > **Machine Learning Model...**.
+1. In the dialog, name the Model Builder project **SentimentAnalysis**, and click **Add**.
+
 ## Choose a scenario
 
 ![Model Builder wizard in Visual Studio](../media/model-builder-scenarios.png)
 
 To train your model, you need to select from the list of available machine learning scenarios provided by Model Builder.
 
-1. In **Solution Explorer**, right-click the *SentimentRazor* project, and select **Add** > **Machine Learning Model...**.
```
Comment (zewditu) [inline docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:69]: ` Machine Learning Model....` it should be ` Machine Learning Model (ML.NET)`

**Comment by JakeRadMSFT** [comment]
Random Search should be under tuners right?

image


https://review.learn.microsoft.com/en-us/dotnet/machine-learning/automated-machine-learning-mlnet?branch=pr-en-us-32011#automl-in-mlnet

**Comment by luisquintanilla** [comment]
@luisquintanilla add GPU recommendations.

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md (hunk 3 above)
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:79]: What's that link?

**Comment by luisquintanilla** [comment]
> Is there a way to download the code samples in the docs?

There will be soon. Here's the PR for it.

dotnet/machinelearning-samples#967

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md (hunk 2 above)
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:70]: Resolved.

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md (hunk 4 above)
Comment (luisquintanilla) [reply docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:69]: This is what I see on the latest version.

image

File: docs/machine-learning/tutorials/sentiment-analysis-model-builder.md (hunk 3 above)
Comment (zewditu) [reply docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:79]: https://learn.microsoft.com/en-us/dotnet/machine-learning/automate-training-with-model-builder#environment

Comment (luisquintanilla) [reply docs/machine-learning/tutorials/sentiment-analysis-model-builder.md:79]: @zewditu Sorry. I'm not sure what the ask is here. Is it that I add this link somewhere in the doc or that we link to it from Model Builder?

**Comment by luisquintanilla** [comment]
@CamSoper this one's ready for review when you get a chance. Looking to merge by tomorrow. Thank you!

**Comment by luisquintanilla** [comment]
```mermaid
flowchart LR
  Input --> Node[Input * Weight + Bias] --> Output
```

**Review by CamSoper — APPROVED** [review]
LGTM @luisquintanilla. I'm approving but please fix the double spaces so the linter is happy.

File: docs/fundamentals/code-analysis/quality-rules/ca2109.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ A public or protected event-handling method was detected.
 
 ## Rule description
 
-An externally visible event-handling method presents a security issue that requires review.
+An externally visible event-handling method presents a security issue that requires review.
```
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply docs/fundamentals/code-analysis/quality-rules/ca2109.md:29]: Thanks. I didn't edit this file, not sure where this came from but I'll make sure it's fixed.

**Pull request merged**